GenderPayGap.WebUI/Scripts/Application/LiveSearch.js

```diff
@@ -202,6 +197,65 @@
     LiveSearch.prototype.collapseFilters = function collapseFilters() {
         $(".govuk-option-select.js-collapsible .js-container-head[aria-expanded=true]").click();
     };
+    
+    LiveSearch.prototype.sendFilterEventToGA = function sendFilterEventToGA() {
+        let selectedFilters = this.state.filter(
+            parameter =>  this.GATrackFilters.filters.map(
+                filter => filter.Group
+            ).includes(parameter.name)
+        );
+        let GAEvent = {
+            hitType: 'event',
+            eventCategory: this.GATrackFilters.category,
+            eventAction: [],
+            eventLabel: {}
+        };
+        sendGpgEvent(this.convertGAEventToHumanReadableFormat(this.addFiltersToGAEvent(selectedFilters, GAEvent)));
+    };
+
+    LiveSearch.prototype.addFiltersToGAEvent = function addFiltersToGAEvent(filters, GAEvent) {
+        filters.forEach(filter => {
+            if (!GAEvent.eventAction.includes(filter.name))
+            {
+                GAEvent.eventAction.push(filter.name);
+                GAEvent.eventLabel[filter.name] = [];
+            }
+            if (!(GAEvent.eventLabel[filter.name].includes(filter.value)))
+            {
+                GAEvent.eventLabel[filter.name].push(filter.value);
+            }
+        });
+        return GAEvent;
+    };
+    
+    LiveSearch.prototype.convertGAEventToHumanReadableFormat = function convertGAEventToHumanReadableFormat(GAEvent) {
+        GAEvent.eventAction = this.convertEventActionToHumanReadableString(GAEvent.eventAction);
+        GAEvent.eventLabel = this.convertEventLabelToHumanReadableString(GAEvent.eventLabel);
+        return GAEvent;
+    };
+
+    LiveSearch.prototype.convertEventActionToHumanReadableString = function convertEventActionToHumanReadableString(eventAction) {
+        return eventAction.map(
+            filterGroup => this.GATrackFilters.filters.find(
+                filter => filter.Group === filterGroup
+            ).Label
+        ).join('; ');
+    };
+
+    LiveSearch.prototype.convertEventLabelToHumanReadableString = function convertEventLabelToHumanReadableString(eventLabel) {
+        let eventLabelString = '';
+        for (const filterGroup in eventLabel)
+        {
+            eventLabelString += this.GATrackFilters.filters.find(filter => filter.Group === filterGroup).Label + ': ';
+            eventLabelString += eventLabel[filterGroup].map(
+                value => this.GATrackFilters.filters.find(
+                    filter => filter.Group === filterGroup).Metadata.find(
+                        filter => filter.Value === value).Label
+            ).join(', ');

```

Comment:
I suggest we move `this.GATrackFilters.filters.find(filter => filter.Group === filterGroup)` to a separate variable. 
I find this quite hard to read. Can you please add some comments? 

Maybe we can build the object we need when iterating through the arrays for the first time in order to have here all the labels. What do you think?

Reply:
I agree it might be easier to do it all in one go. I'll try this :).

Reply:
I've completely refactored the filter event creation code. I think it's way more readable now. Please let me know what you think 👍.

Reply:
It is indeed, thank you!